unity-game/Assets/Scripts/ElevatorScripts/AutoElevator.cs

```diff
@@ -0,0 +1,220 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class AutoElevator : MonoBehaviour
+{
+    public bool notMoving;
+    public bool reachedDestination;
+    public bool onStartPoint;
+    public bool onEndPoint;
+
+    public GameObject objectToMove;
+    public GameObject theRightWall;
+    public GameObject theLeftWall;
+    public GameObject theStartLeftBarrier;
+    public GameObject theStartRightBarrier;
+    public GameObject theEndLeftBarrier;
+    public GameObject theEndRightBarrier;
+    public GameObject theStartPoint;
+    public GameObject theEndPoint;
+
+    public Transform startPoint;
+    public Transform endPoint;
+
+    public float moveSpeed;
+    public float waitTime;
+    public float waitCounter;
+    
+    private Vector3 currentTarget;
+
+
+    void Start()
```

Comment:
this might be better in `Awake`. since `Start` would get called again in `OnEnable` and this would break the current state if the platform was not at the start position. (Will need to handle disable/enable in `Start` too)